crates/astria-core/src/primitive/v1/asset/denom.rs

```diff
@@ -634,17 +634,14 @@ mod tests {
             TooManySegments,
         };
         #[track_caller]
-        // allow: silly lint
-        #[allow(clippy::needless_pass_by_value)]
+        #[expect(clippy::needless_pass_by_value, reason = "silly lint")]
```

Comment:
I wrote this, it's not good.
```suggestion
        #[expect(
            clippy::needless_pass_by_value,
            reason = "asserting on owned variants is less noisy then passing them by reference"
        )]